django/contrib/staticfiles/finders.py

```diff
@@ -113,17 +113,22 @@ def check(self, **kwargs):
                 )
         return errors
 
-    def find(self, path, all=False):
+    def find(self, path, fetch_all=False, **kwargs):
```

Comment:
I'm not familiarized in how deprecations are handled in python/django, but I think that adding `**kwargs` is overkill. Maybe add something like:

```python
    def find(self, path, fetch_all=False, all=None):
        if all is not None:
            # TODO: Add a warning or something
            fetch_all = all
```

This sadly doesn't fix shadowing `all`, but by adding a deprecation warning it will allow us to remove it in future versions.

Maybe someone can give us some advice on how to handle this?

Reply:
Thanks for the feedback @EnriqueSoria. As per your suggestion I've added the deprecations warning for the use of the `all` param in 03b3b1faabd1970ad846f1266ffb320b1ce0508b . On the other hand, I don't see why the use of `**kwargs` is an overkill in this case. I think it fits the case, removing the use of the builtin `all` function and at the same time avoiding the errors for third party packages that may use or extend the existing finders API.

Reply:
When using a function that has `**kwargs` you don't know which parameters it does accept. That's why I dislike using it. 

But I guess in this case is worth.

Reply:
Using `**kwargs` has the undesired side effect of allowing for arbitrary named arguments and the silently ignoring them. This is something that we should avoid so I see two options:
1. Continue using `all` for the deprecation period, is not that terrible since we have already done so for the last 10+ years. 
2. Change to use `**kwargs` but then we should check that no other named param was given and raise the corresponding `TypeError`.

I think 1 is the best option, so I would suggest to go down that path.

Reply:
@avallbona The comment above is what I was referring in the global comment.

Reply:
Sorry I missed this comment 😬 . I'm not sure I like the option 1, we don't solve the problem right away and we pospone the solution to a future version. So, In option 1, would we: keep the the method signature as `find(path, find_all=False, all=False)` or just like `find(path, all=False)` and raise the deprecation warning?

Reply:
@nessita Addressed the issue in eacad9336b5225b6bd47894cd1d7d5db33367a47